bundles/org.openhab.binding.teslascope/src/main/java/org/openhab/binding/teslascope/internal/TeslascopeHandler.java

```diff
@@ -0,0 +1,381 @@
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        schedulePoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        logger.debug("Scheduling poll every {} s", config.refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::pollStatus, 0, config.refreshInterval,
+                TimeUnit.SECONDS);
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() {
+        DetailedInformation detailedInformation = null;
+
+        try {
+            detailedInformation = webTargets.getDetailedInformation(config.publicID, config.apiKey);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (TeslascopeAuthenticationException e) {
+            logger.debug("Unexpected authentication error connecting to Teslascope API", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        } catch (TeslascopeCommunicationException e) {
+            logger.debug("Unexpected error connecting to Teslascope API", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+        updateState(TeslascopeBindingConstants.CHANNEL_VIN, new StringType(detailedInformation.vin));
+        updateState(TeslascopeBindingConstants.CHANNEL_VEHICLE_NAME, new StringType(detailedInformation.vehicleName));
+        updateState(TeslascopeBindingConstants.CHANNEL_VEHICLE_STATE, new StringType(detailedInformation.vehicleState));
+        updateState(TeslascopeBindingConstants.CHANNEL_ODOMETER,
+                new QuantityType<>(detailedInformation.odometer, ImperialUnits.MILE));
+
+        // charge state
+        updateState(TeslascopeBindingConstants.CHANNEL_BATTERY_LEVEL,
+                new DecimalType(detailedInformation.batteryLevel));
+        updateState(TeslascopeBindingConstants.CHANNEL_USABLE_BATTERY_LEVEL,
+                new DecimalType(detailedInformation.usableBatteryLevel));
+        updateState(TeslascopeBindingConstants.CHANNEL_BATTERY_RANGE,
+                new QuantityType<>(detailedInformation.batteryRange, ImperialUnits.MILE));
+        updateState(TeslascopeBindingConstants.CHANNEL_ESTIMATED_BATTERY_RANGE,
+                new QuantityType<>(detailedInformation.estBatteryRange, ImperialUnits.MILE));
+        // charge_enable_request isn't the right flag to determine if car is charging or not
+        if (detailedInformation.chargingState.equals("Charging")) {
+            updateState(TeslascopeBindingConstants.CHANNEL_CHARGE, OnOffType.ON);
+        } else {
+            updateState(TeslascopeBindingConstants.CHANNEL_CHARGE, OnOffType.OFF);
+        }
```

Comment:
```suggestion
            updateState(TeslascopeBindingConstants.CHANNEL_CHARGE, OnOffType.from("Charging".equals(detailedInformation.chargingState)));
```

Reply:
I knew there had to be a better way to do this :)